lib/Sema/CSDiag.cpp

```diff
@@ -6686,8 +6686,14 @@ static bool diagnoseKeyPathComponents(ConstraintSystem &CS, KeyPathExpr *KPE,
                                corrections);
 
       if (currentType)
-        TC.diagnose(componentNameLoc, diag::could_not_find_type_member,
-                    currentType, componentName);
+        if (currentType->is<TupleType>()) {
```

Comment:
Just FYI but you can do `isa<TupleType>(currentType.getPointer())` e.g. that's the only check for presence of `ParenType`.

Reply:
Oh, thanks for the tip! 😄